### ISSUE TITLE
Maven phases are listed alphabetically, not in execution order

### ISSUE THREAD

**Issue by kriegaex** [grisha9/gmaven-plugin#9]
As a user, I expect the Maven phases to be listed
  * grouped by lifecycle (e.g. clean, default, site lifecycles),
  * per lifecycle ordered by Maven phase execution order.

Instead, I see this:

![image](https://github.com/grisha9/gmaven-plugin/assets/1537384/0e48e3bf-7a32-4f4b-b93d-74143308b356)

You see that e.g. _validate_ is not before _compile_ where it belongs but before _verify_.

**Comment by kriegaex** [comment]
Additional problem: Depending on which order I click to select multiple goals in, they are executed in that order, e.g. if I select _package_, then _clean_, running the phases will happen in this order, i.e. Maven will first package a mudule and subsequently clean it, which does not make any sense. The built-in packager executes phases in the expected order per lifecycle.

One more thing: There is no "run" button like in the built-in Maven plugin, i.e. I always need to right-click and use the pop-up menu to execute a build with more than one phase.

**Comment by grisha9** [comment]
I use [External System API](https://plugins.jetbrains.com/docs/intellij/external-system-integration.html) for integration with Maven. 
I simply prepare the data, and the platform handles the display. And at the moment there is no option to change the sorting. I created the issue about this: https://youtrack.jetbrains.com/issue/IDEA-351908/

Execution order will be fixed.

**Comment by grisha9** [comment]
Task execution order fixed in https://github.com/grisha9/gmaven-plugin/issues/6
> Maven phases are listed alphabetically
It is platform problem. I created the issue about this: https://youtrack.jetbrains.com/issue/IDEA-351908/

**Comment by grisha9** [comment]
Yes you are right. I didn’t specify - Maven, as one of the very first plugins, has its own customized tool window. And there they implement all the logic they want.
But then they generalized this approach and created an [External System API](https://plugins.jetbrains.com/docs/intellij/external-system-integration.html) for external systems. This api is used by later plugins such as Gradle and Sbt. (Maven uses it only partially and not for tool windows). The main essence of this approach is that plugins for working with external systems simply prepare data (get it from build system). And ready-made UI components from the platform allow you to display this data and work with it (import into the project structure/run tasks)
And I really don’t want to write my own custom tool window. I think it would be more correct to submit a request for improvement.
